[hyc17003] iP

hyc17003 opened the pull request:
# ChatterBot

This is a Chatterbot to help you manage your tasks. Given below are instructions on how to use it.

> Why couldn't the computer take its hat off? Its caps lock was on!

### ChatterBot can:
- add your tasks to a list
- display the task list when requested
- delete tasks from the task list
- mark tasks as done or undone
- search for task(s) based on a specific keyword

### Here are the steps to using it:
1. Download ChatterBot [here](https://github.com/hyc17003/ip)
2. Double-click it
3. Add your tasks
4. Watch as task-managing becomes ~~easier~~ ***so much*** easier!!! 😉

You can use it for:
- [x] Schoolwork
- [x] Chores
- [x] _anything, really_

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Here's what you'll be greeted by upon starting up ChatterBot: 
`System.out.println("Hello! I'm " + logo + "\nWhat can I do for you?");`

What are you waiting for? Use it now!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

hyc17003 committed:
Add ChatterBot.java

hyc17003 committed:
Echo ChatterBot

hyc17003 committed:
Add, List ChatterBot

hyc17003 committed:
Mark as Done ChatterBot

hyc17003 committed:
ToDos, Events, Deadlines ChatterBot

hyc17003 committed:
ToDos, Events, Deadlines ChatterBot

hyc17003 committed:
A-TextUiTesting

hyc17003 committed:
ToDos, Events, Deadlines ChatterBot

hyc17003 committed:
Handle Errors ChatterBot

hyc17003 committed:
Delete ChatterBot

hyc17003 committed:
Save ChatterBot

hyc17003 committed:
Implement Dates feature for ChatterBot

hyc17003 committed:
Merge branch 'branch-Level-8'
Implement Date feature for ChatterBot

hyc17003 committed:
Make the code more OOP

hyc17003 committed:
Merge branch 'branch-A-moreOOP'
Make the code more OOP

hyc17003 committed:
Organise into packages

hyc17003 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

hyc17003 committed:
Merge branch 'branch-A-Gradle'

Set up Gradle.

xenosf commented on src/main/java/chatterbot/ui/Ui.java:
I like how clear the method names are. Perhaps adding header comments to public methods could make it extra clear?

xenosf commented on src/main/java/chatterbot/parser/Parser.java:
```diff
@@ -0,0 +1,105 @@
+
+public class Parser {
+
+    public static void evaluateCommand(String userMessage, Ui ui, ArrayList<Task> list, Storage storage, String file, TaskList taskList) {
```
This method is very long -- would it make the code clearer to split it up into smaller methods? You can consider creating a Command class as outlined in the [A-MoreOOP](https://nus-cs2103-ay2324s1.github.io/website/schedule/week3/project.html#a-moreoop) extension stretch goal to separate the command parsing and execution.

xenosf commented on src/main/java/chatterbot/data/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+
+public class Deadline extends Task {
+
+    protected String by;
```
Could this variable name be more descriptive? Maybe consider something like "deadline"

xenosf commented on src/main/java/chatterbot/data/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+        return ld.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+    }
+
+    @Override
```
Could the method name be clearer if it was named as a verb? Maybe something like `convertToFileText`

xenosf commented on src/main/java/chatterbot/data/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + modifyBy(by) + ")";
+    }
+
```
I'm not sure what this method is supposed to do based on the name. Is it to convert the "by" string to a LocalDate? If it is, you can consider changing the name to something like `convertByToLocalDate`

hyc17003 committed:
Add JUnit tests for Deadline, Todo and Event

hyc17003 committed:
Edit to adhere to coding standards

hyc17003 committed:
Added JavaDocs

hyc17003 committed:
Adjusted to adhere to coding standards

hyc17003 committed:
Make changes to rectify build error

hyc17003 committed:
Make changes to rectify bug.

hyc17003 committed:
Add method that gives users a way to find a task by searchingfor a keyword.

hyc17003 committed:
Merge branch 'branch-A-JavaDoc', which adds JavaDoc comments to the entire code.

hyc17003 committed:
Resolve merge conflicts between master and branch-A-CodingStandard

hyc17003 committed:
Fixedv variable name

hyc17003 committed:
Add method that gives users a way to find a task by searching for a keyword.

hyc17003 committed:
Update README.md

hyc17003 committed:
Update README.md

hyc17003 committed:
Add GFMD in iP description

hyc17003 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	README.md

hyc17003 committed:
Revert ChatterBot.class back to previous state

hyc17003 committed:
Add GUI

hyc17003 committed:
Add GUI

hyc17003 committed:
Push branch with tag

hyc17003 committed:
Merge branch 'Use_GFMD'

# Conflicts:
#	data/ChatterBot.txt

hyc17003 committed:
GUI: Add functionality for list to read and be initialised from existing file.

The list starts off as empty, and does not read from any file.

The data has to be added for each time ChatterBot is run, for the GUI.

Let's get the list to read the ChatterBot.txt file and be initialised with the current values inside.

Using this will allow the user to use ChatterBot over a period of time with the same tasks inside, instead of having to re-add them every time.

Still some bugs to fix, like for the 'delete' and 'mark'/'unmark' functions.

hyc17003 committed:
Add assertions for Parser, Task and TaskList classes.

There are no assertions in ChatterBot.

Assertions can detect possible bugs, and the sooner execution stops, the safer it is.

Assertions are added for some classes.

They will make sure the code behaves in the way it should be.

hyc17003 committed:
Add more assertions

hyc17003 committed:
Improve code quality

hyc17003 committed:
Merge pull request #2 from hyc17003/branch-A-Assertions

Add more assertions

hyc17003 committed:
Resolve merge conflict

hyc17003 committed:
Merge pull request #3 from hyc17003/branch-A-CodeQuality

Improve code quality

hyc17003 committed:
Improve code OOP

hyc17003 committed:
Resolve merge conflicts

hyc17003 committed:
Improve code OOP

hyc17003 committed:
Add C - Detect Duplicates Extension

hyc17003 committed:
Add more JavaDocs and improve code standards

hyc17003 committed:
Improved the GUI

hyc17003 committed:
Update README and fix delete bug

hyc17003 committed:
Add more unit testing

hyc17003 committed:
Update README.md

hyc17003 committed:
Update README.md

hyc17003 committed:
Update User Guide to include link to download JAR file

hyc17003 committed:
Merge branch 'master' of https://github.com/hyc17003/ip
Update User Guide to include link to JAR file

hyc17003 committed:
Add exceptions handling

hyc17003 committed:
Edit utility code

hyc17003 committed:
Edit duplicate function

hyc17003 committed:
Edit file accessing feature

hyc17003 committed:
Edit file accessing feature
